### PR TITLE
Indicate that abs() method accept argument that implement __abs__(), …

### PR DESCRIPTION
**Pull request by Windsooon**
…just like call() method in the docs

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


When your account is ready, please add a comment in this pull request
and a Python core developer will remove the `CLA not signed` label
to make the bot check again.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by Windsooon** [comment]
I signed the CLA, Thank you for reviewing.

**Comment by eric-wieser** [comment]
Typo in the commit message - `add` -> `abs`

**Comment by Windsooon** [comment]
@eric-wieser Thanks, I fixed it now.

**Review by jdemeyer — CHANGES_REQUESTED** [review]
I think the wording could be improved: what does "The argument may be an integer or a floating point number." really mean now? You should explain that this applies to integers, floating point numbers, complex numbers and anything defining `__abs__`.

File: Doc/library/functions.rst
Comment (JulienPalard) [inline Doc/library/functions.rst]: There's a capital letter after a coma.
Also I have another proposition about this coma:

> [...] integer, a floating point number, a complex number, or any object defining the :meth:`__abs__` method.  If the argument is a complex number, its magnitude is returned.

I agree with @jdemeyer, it could be "simplified" even to ```Accept any object *x* implementing :meth:`__abs__` ```, but this documentation is not the language reference, its audience is newcomers, they care about being able to pass the three builtins they just learnt `int`, `float`, `complex` like `abs(-5)` and `abs(-5.5)`.

**Review by JulienPalard — CHANGES_REQUESTED** [review]

**Comment by csabella** [comment]
@Windsooon, please resolve the merge conflict and address the code review comments.  Thank you!

**Comment by eric-wieser** [comment]
> Typo in the commit message - `add` -> `abs`

Typo is still present in the PR title

**Comment by Windsooon** [comment]
To make it easier, how about

"... integer or a floating point number or anything defining __abs__.  If the argument is a complex number, its magnitude is returned.

**Comment by eric-wieser** [comment]
PR title now says `add`, but this patch is about `abs`?

**Comment by Windsooon** [comment]
Sorry for the mistake. I just fixed it.

**Comment by JulienPalard** [comment]
> To make it easier, how about
> 
>     "... integer or a floating point number or anything defining __abs__. If the argument is a complex number, its magnitude is returned."

Yes, by using

    :meth:`__abs__`

instead of a raw __abs__.

I'm not a native english speaker but I think the first or is now redundent, what about:

    The argument may be an integer, a floating point number, or anything defining :meth:`__abs__`.

I'd maybe prefer a bit `object` over `anything`, like:

    The argument may be an integer, a floating point number, or an object implementing :meth:`__abs__`.

**Comment by Windsooon** [comment]
@JulienPalard  Thank you. I updated the PR based on your suggestion.

**Comment by JulienPalard** [comment]
@Windsooon could you resolve the conflict?
I would personally do it using `git fetch upstream; git rebase upstream/master` (my python/cpython remote being named `upstream`), as your PR is based on a commit from 2018.

**Comment by csabella** [comment]
@Windsooon, please take a look at the last review comments.  Thank you!

**Comment by terryjreedy** [comment]
rebase commonly messes up branches and PRs with the cpython workflow.  Did you do git merge in master before updating the branch?  git merge upstream/master usually works.  There might be a merge conflict, but not likely with a narrow PR like this one.  You could still try merge on your local branch if not deleted yet.

**Comment by Windsooon** [comment]
@terryjreedy Thank you. I just delete the old branch. However, I don't know why my PR will conflict with the current master branch. I will check it out later.

**Pull request closed without merging**
